src/main/java/com/treasuredata/android/cdp/SegmentResultsOfException.java

```diff
@@ -0,0 +1,15 @@
+package com.treasuredata.android.cdp;
+
+public class SegmentResultsOfException extends SegmentsResult {
```

Comment:
Should this SegmentResultsOfException and SegmentsResultOfJSON not be public?

Reply:
Also, can we generalize SegmentResultsOfException and  SegmentsResultsOfJSONObject with error format to something like CDPAPIException and CDPResponseException? I think these two case have nothing to do with SegmentsResults. The other two cases (SegmentsResultsOfJSONArray and SegmentsResultsOfJSONObject without an error format) can be extended from a new abstract class like CDPResponseJSON.
This would help tremendously future new api (for example fetchRealtimeUserSegments).
Of course we can always refactor later, it's up to you :) Just my 2 cents

Reply:
I renamed to ErrorLookupResult and JSONObject/ArrayLookupResult, since the old terms sound confusing, there's not always segments data there.
The JSONLookupResult or ErrorLookupResult is not meant to be exceptions, they all (including JSONArrayLookupResult) are just intermediate encapsulation to able to know to invoke the callbacks. They hide the logic of how the callbacks get handled (either onSuccess or onCallback get called, although it's pretty obvious in the ErrorLookupResult name). The suggested way expose or at least imply that logic outside.